src/core/ddsc/src/dds_read.c

```diff
@@ -13,6 +13,7 @@
 #include <string.h>
 #include "dds__entity.h"
 #include "dds__reader.h"
+#include "dds__writer.h"
```

Comment:
I have a bit of a problem with retaining `dds_return_loan` as a function in a file so strongly associated with reading data when after extending it to also deal with loans on writers. I think it would make more sense to move the dispatching code to `dds_entity.c` and keep only the reader-specific parts here, for example in a function `dds_return_reader_loan`.

Reply:
Alternatively the loaning code could get its on compilation unit and would contain functions for returning loans for different entities (general and specific). We lose the de facto object oriented notion then if we move reader and writer specific implementations also there. Furthermore the functions to obtain the loan would be elsewhere, which is also confusing.